File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        //use an int array
        std::array<int,26> cmap;
        
        int l = 0, r = 0, m = 0, res = 0;
        
        for (r; r < s.size(); ++r) {
            ++cmap[s[r] - 'A'];
            //Find if current character is the longest character in our window
            m = max(m, cmap[s[r] - 'A']);
            // Find if replace the longest character in our window m
            // still within our budget of k chacters
            while (r - l + 1 - m > k) {
                --cmap[s[l] - 'A'];
                ++l;
            }
            res = max(res, r - l + 1);
        }
        
        return res;
    }
};

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.go
func characterReplacement(s string, k int) int {
    l := 0
    w_l := 0
    w_c := 0
    veca := [26]int{};
    
    for r := 0; r < len(s); r++ {
        veca[s[r] - 'A']++;
        w_c = max(w_c, veca[s[r] - 'A']);
        if (r - l + 1 - w_c > k) {
            veca[s[l] - 'A']--;
            l++;
        }
        w_l = max(w_l, r - l + 1);
    } 
    return w_l;
}